test/end2end_test.go

```diff
@@ -6855,8 +6884,14 @@ func (s) TestHTTPHeaderFrameErrorHandlingNormalTrailer(t *testing.T) {
 			},
 			errCode: codes.Internal,
 		},
-	} {
-		doHTTPHeaderTest(t, test.errCode, test.responseHeader, test.trailer)
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
+1 for using `t.Run()` here.

Reply:
I think there are more places to use `t.Run()` in this package. But seemed like a lot of diff for this PR